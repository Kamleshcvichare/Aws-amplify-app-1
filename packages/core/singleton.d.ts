export * from './lib-esm/Singleton';