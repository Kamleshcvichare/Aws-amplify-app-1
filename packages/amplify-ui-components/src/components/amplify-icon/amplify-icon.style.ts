import { css } from 'emotion';

export const icon = css`
  fill: var(--deep-squid-ink);
`;