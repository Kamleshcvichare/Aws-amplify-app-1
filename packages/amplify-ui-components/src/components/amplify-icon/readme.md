# amplify-icon



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description                                              | Type                                                                             | Default     |
| --------------- | ---------------- | -------------------------------------------------------- | -------------------------------------------------------------------------------- | ----------- |
| `name`          | `name`           | (Required) Icon name used to determine the icon rendered | `"enter-vr" \| "exit-vr" \| "maximize" \| "minimize" \| "sound" \| "sound-mute"` | `undefined` |
| `overrideStyle` | `override-style` |                                                          | `boolean`                                                                        | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
