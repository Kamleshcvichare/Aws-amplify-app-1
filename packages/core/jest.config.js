module.exports = {
	...require('../../jest.config'),
	coverageThreshold: {
		global: {
			branches: 74, // todo(ashwinkumar6): temp, add tests and bump back
			functions: 72,
			lines: 87, // todo(ashwinkumar6): temp, add tests and bump back
			statements: 88,
		},
	},
};
