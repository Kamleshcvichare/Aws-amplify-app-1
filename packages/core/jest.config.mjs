import { getJestConfig } from '../../jest/getJestConfig.mjs';
import tsconfig from './tsconfig.json' assert { type: 'json' };

const jestConfig = getJestConfig(tsconfig.compilerOptions, {
	coverageThreshold: {
		global: {
			branches: 75,
			functions: 72,
			lines: 88,
			statements: 88,
		},
	},
});

export default jestConfig;
