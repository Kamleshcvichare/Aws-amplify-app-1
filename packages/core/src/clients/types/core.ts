/**
 * General None HTTP-specific request interface
 */
export interface Request {
	url: URL;
	body?: unknown;
}

export interface Response {
	body: unknown;
}

export interface TransferHandler<
	Input extends Request,
	Output extends Response,
	TransferOptions
> {
	(request: Input, options: TransferOptions): Promise<Output>;
}

/**
 * A slimmed down version of the AWS SDK v3 middleware handler, only handling instantiated requests
 */
export type MiddlewareHandler<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (request: Input, options: MiddlewareOptions) => Promise<Output>;

export type MiddlewareContext = Record<string, unknown>;

/**
 * A slimmed down version of the AWS SDK v3 middleware, only handling tasks after Serde.
 */
export type Middleware<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (
	next: MiddlewareHandler<Input, Output, MiddlewareOptions>,
	context: MiddlewareContext
) => MiddlewareHandler<Input, Output, MiddlewareOptions>;
