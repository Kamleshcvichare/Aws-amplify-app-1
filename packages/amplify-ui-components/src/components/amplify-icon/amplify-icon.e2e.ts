import { newE2EPage } from '@stencil/core/testing';

describe('amplify-icon', () => {
  it('renders with a different test name', async () => {
    const page = await newE2EPage();

    await page.setContent(
      `<amplify-icon name='sound-mute'></amplify-icon>`,
    );
    const screenshot = await page.compareScreenshot('Amplify icon', {fullPage: true});
    expect(screenshot).toMatchScreenshot({ allowableMismatchedPixels: 10 });

    const iconElement = await page.find('amplify-icon');
    expect(iconElement).not.toBeNull();
  });
});