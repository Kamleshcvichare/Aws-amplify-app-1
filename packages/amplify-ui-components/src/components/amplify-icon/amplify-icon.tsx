import { Component, Prop, Watch, h } from '@stencil/core';
import { icon } from './amplify-icon.style';
import { styleNuker } from '../../common/helpers';
import { AMPLIFY_UI_ICON } from '../../common/constants';

export type IconNameType = 'sound-mute' | 'sound' | 'maximize' | 'minimize' | 'enter-vr' | 'exit-vr';

import { icons } from './icons';

@Component({
  tag: 'amplify-icon',
  shadow: false,
})
export class AmplifyIcon {
  /** (Required) Icon name used to determine the icon rendered */
  @Prop() name: IconNameType;
  @Prop() overrideStyle: boolean;

  @Watch('name')
  validateName(newValue: string) {
    const isBlank = typeof newValue == null;
    if (isBlank) { throw new Error('name: required') };
  }

  render() {
    return (
      <span class={styleNuker(this.overrideStyle, AMPLIFY_UI_ICON, icon)}>
        {icons[this.name]}
      </span>
    );
  }
}
