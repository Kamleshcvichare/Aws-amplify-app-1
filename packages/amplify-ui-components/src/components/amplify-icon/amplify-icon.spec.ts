import { newSpecPage } from '@stencil/core/testing';
import { AmplifyIcon } from './amplify-icon'

describe('amplify-icon', () => {
  it('renders correct HTML', async () => {
    const page = await newSpecPage({
      components: [AmplifyIcon],
      html: `<amplify-icon name='sound-mute'></amplify-icon>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});