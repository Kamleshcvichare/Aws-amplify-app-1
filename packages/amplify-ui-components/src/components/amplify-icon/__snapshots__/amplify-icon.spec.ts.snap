// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-icon renders correct HTML 1`] = `
<amplify-icon name="sound-mute">
  <span class="amplify-ui-icon css-11j4bfh">
    <svg height="19" viewBox="0 0 19 19" width="19" xmlns="http://www.w3.org/2000/svg">
      <g fillRule="evenodd" id="icons/minis/volumeOff">
        <path d="M3.48026899,12.9630494 C3.63825091,12.9630494 3.79237961,13.0108921 3.92264322,13.1003479 L8.77467683,16.8113609 C9.29423971,17.1679383 10,16.7950396 10,16.1637406 L10,3.78619489 C10,3.15489596 9.29423971,2.78199725 8.77467683,3.13857463 L3.92264322,6.84545211 C3.79237961,6.93490793 3.63825091,6.9827506 3.48026899,6.9827506 L1.78294894,6.9827506 C1.3505185,6.9827506 1,7.33409518 1,7.76754476 L1,12.1781306 C1,12.6117048 1.3505185,12.9630494 1.78294894,12.9630494 L3.48026899,12.9630494 Z M17.2118156,7 L15.0918385,9.11997713 L12.9718614,7 L12,7.97174685 L14.1200917,10.091724 L12,12.2118156 L12.9718614,13.1835625 L15.0918385,11.0635854 L17.2118156,13.1835625 L18.1835625,12.2118156 L16.0635854,10.091724 L18.1835625,7.97174685 L17.2118156,7 Z" id="Fill-2"></path>
      </g>
    </svg>
  </span>
</amplify-icon>
`;
